pkgs/build-support/rust/carnix.nix

```diff
@@ -461,690 +456,1053 @@ rec {
-      }))
-    [ libc_0_2_33_features cc_1_0_3_features ];
-  bitflags_0_7_0 = f: bitflags_0_7_0_ rec {};
-  bitflags_0_7_0_features = features:
-    lib.lists.foldl' (features: f: f features)
-      (lib.attrsets.recursiveUpdate features (rec {
-      }))
-    [ ];
-  bitflags_1_0_1 = f: bitflags_1_0_1_ rec {
-    features = mkFeatures (f.bitflags_1_0_1 or {});
-  };
-  bitflags_1_0_1_features = features:
-    lib.lists.foldl' (features: f: f features)
-      (lib.attrsets.recursiveUpdate features (rec {
-        bitflags_1_0_1."example_generated".self_default = hasFeature (bitflags_1_0_1.default or {}) || hasFeature (features.bitflags_1_0_1.default or {});
-      }))
-    [ ];
-  carnix_0_6_0 = f: carnix_0_6_0_ rec {
-    dependencies = [ (clap_2_28_0 f) (env_logger_0_4_3 f) (error_chain_0_11_0 f) (itertools_0_7_3 f) (log_0_3_8 f) (nom_3_2_1 f) (regex_0_2_2 f) (rusqlite_0_13_0 f) (serde_1_0_21 f) (serde_derive_1_0_21 f) (serde_json_1_0_6 f) (tempdir_0_3_5 f) (toml_0_4_5 f) ];
-  };
-  carnix_0_6_0_features = features:
-    lib.lists.foldl' (features: f: f features)
-      (lib.attrsets.recursiveUpdate features (rec {
-        clap_2_28_0.default.from_carnix_0_6_0__default_ = true;
-        env_logger_0_4_3.default.from_carnix_0_6_0__default_ = true;
-        error_chain_0_11_0.default.from_carnix_0_6_0__default_ = true;
-        itertools_0_7_3.default.from_carnix_0_6_0__default_ = true;
-        log_0_3_8.default.from_carnix_0_6_0__default_ = true;
-        nom_3_2_1.default.from_carnix_0_6_0__default_ = true;
-        regex_0_2_2.default.from_carnix_0_6_0__default_ = true;
-        rusqlite_0_13_0.default.from_carnix_0_6_0__default_ = true;
-        serde_1_0_21.default.from_carnix_0_6_0__default_ = true;
-        serde_derive_1_0_21.default.from_carnix_0_6_0__default_ = true;
-        serde_json_1_0_6.default.from_carnix_0_6_0__default_ = true;
-        tempdir_0_3_5.default.from_carnix_0_6_0__default_ = true;
-        toml_0_4_5.default.from_carnix_0_6_0__default_ = true;
-      }))
-    [ clap_2_28_0_features env_logger_0_4_3_features error_chain_0_11_0_features itertools_0_7_3_features log_0_3_8_features nom_3_2_1_features regex_0_2_2_features rusqlite_0_13_0_features serde_1_0_21_features serde_derive_1_0_21_features serde_json_1_0_6_features tempdir_0_3_5_features toml_0_4_5_features ];
-  cc_1_0_3 = f: cc_1_0_3_ rec {
+      ++ (if features.backtrace_0_3_4."dbghelp-sys" or false then [(dbghelp_sys_0_2_0 { features = features; })] else [])
+      ++ (if features.backtrace_0_3_4."kernel32-sys" or false then [(kernel32_sys_0_2_2 { features = features; })] else [])
+      ++ (if features.backtrace_0_3_4."winapi" or false then [(winapi_0_2_8 { features = features; })] else []) else []);
+    features = mkFeatures (features.backtrace_0_3_4 or {});
+  };
+  backtrace_0_3_4_features = f:
+    builtins.deepSeq f (lib.lists.foldl' (features: fun: fun features)
+      (lib.attrsets.recursiveUpdate f (rec {
+        backtrace_0_3_4."addr2line" =
```

Comment:
is this equivalent to?

```nix
backtrace_0_3_4."addr2line" = (f.backtrace_0_3_4."addr2line" or
  f.backtrace_0_3_4."gimli-symbolize" or
  backtrace_0_3_4."gimli-symbolize" or false);
```

Reply:
It would be great, but it doesn't look like it's equivalent:
```
nix-repl> { a = false; }.a or { a = true; }.a
false
```